engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/MergeStatisticsStep.java

```diff
@@ -120,8 +129,11 @@ protected ExecuteResult doWork(ExecutableContext context) throws ExecuteExceptio
                         tempFile.delete();
                 }
             }
+            sourceRecordCount *= effectiveTimeRange == 0 ? 0 : newSegment.getTSRange().duration() / effectiveTimeRange;
```

Comment:
why not convert to double to calculate the accurate data?

Reply:
It's not necessary to do so. Row count itself is an approximate one. And this data is set to long type at other place and we shouldn't break the compatibility.

Reply:
For details, please check http://issues.apache.org/jira/browse/KYLIN-3453